public/systems_monitoring_graphs.js

```diff
@@ -0,0 +1,578 @@
+                .attr("height", Height + Margin.top + Margin.bottom)
+                .append("g")
+                .attr("transform",
+                    "translate(" + Margin.left + "," + Margin.top + ")");
+
+            // Add X axis
+            let x = d3.scaleTime().domain(d3.extent(data, d => new Date(d.datetime))).range([1, Width]);
+            let xAxis = svg.append("g")
+                .attr("transform", "translate(0," + Height + ")")
+                .call(d3.axisBottom(x).ticks(d3.timeDay.every(1)).tickFormat(dayTimeFormat));
+
+            // Add the X gridlines
+            svg.append("g")			
+                .attr("class", "memoryXGrid")
+                .attr("transform", "translate(0," + Height + ")")
+                .call(d3.axisBottom(x)
+                    .tickSize(-Height)
+                    .tickFormat("")
+                )
+
+            // Rotate axis ticks
+            xAxis.selectAll("text")
+                .style("text-anchor", "end")
+                .attr("dx", "-.8em")
+                .attr("dy", ".15em")
+                .attr("transform", "rotate(-65)");
+
+            // Add X axis label
+            svg.append("text")
+                .attr("text-anchor", "end")
+                .attr(
+                    "transform",
+                    "translate(" + Width / 2 + " ," + (Height + Margin.top + 50) + ")"
+                )
+                .style("text-anchor", "middle")
+                .text("Date (dd:hh:m)");
+
+            // Add Y axis label
+            svg.append("text")
+                .attr("transform", "rotate(-90)")
+                .attr("y", 0 - Margin.left)
+                .attr("x", 0 - Height / 2)
+                .attr("dy", "1em")
+                .style("text-anchor", "middle")
+                .text("Memory Utilization(GB)");
+
+            let decimalFormatter = d3.format(".2s");
+            let yLimit = data[0].memory_usage.total;
+            // Add Y axis
+            let y = d3.scaleLinear()
+                .domain([0, yLimit])
+                .range([ Height, 0 ]);
+            svg.append("g").call(d3.axisLeft(y).ticks(5).tickFormat((d) => decimalFormatter(d).replace('G', 'GB')))
+
+            // Add the Y gridlines
+            svg.append("g")			
+                .attr("class", "memoryGrid")
+                .call(d3.axisLeft(y)
+                    .tickSize(-Width)
+                    .tickFormat("")
+                )
+
+            // Add a clipPath: everything out of this area won't be drawn.
+            let clip = svg.append("defs").append("svg:clipPath")
+                .attr("id", "clip")
+                .append("svg:rect")
+                .attr("width", Width )
+                .attr("height", Height )
+                .attr("x", 1)
+                .attr("y", 0);
+
+            // Add brushing
+            let brush = d3.brushX()                 // Add the brush feature using the d3.brush function
```

Comment:
Can you move this to be next to where you use it (L295)? It's a bit easier to read if the declaration and use are nearby

Reply:
Moved all the brush variable declaration next to where its used, thanks.